website/docs/r/rde.html.markdown

```diff
@@ -0,0 +1,199 @@
+`input_entity` and `input_entity_url` will only specify the RDE contents that were used either on creation or in a deliberate
+update that will cause the RDE contents to be **completely overridden**.
+
+As per this last point, one needs to be careful when updating `input_entity` or `input_entity_url`, as Terraform will apply
+whatever changes were done, ignoring the real state from `computed_entity`. To perform a real update, one
+needs to check the contents of the `computed_entity` and do some diff with the original input.
+
+In other words:
+
+~> When you want to update an RDE and `entity_in_sync` is `false`, you should always merge the contents
+of `computed_entity` and `input_entity` to avoid overriding the whole entity by mistake with an old value. 
+
+## RDE resolution
+
+When a RDE is created, its `state` will be `PRE_CREATED`, which means that the entity JSON was not validated against the
+[RDE Type](/providers/vmware/vcd/latest/docs/resources/rde_type) schema. After resolution, `state` should be either `RESOLVED`
+or `RESOLUTION_ERROR` if the input JSON doesn't match the schema.
+
+The RDE must be eventually resolved to be used or deleted, and this operation can be done either by Terraform with
+`resolve=true`, or by a 3rd party actor that will do it behind the scenes at some point (in this case, the Terraform resource
+should have `resolve=false` to avoid being resolved).
+In this last scenario, it is advisable to mark `resolve_on_removal=true` so Terraform can delete the RDE even if it was not
+resolved by anyone.
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional) Name of the [Organization](/providers/vmware/vcd/latest/docs/resources/org) that will own the RDE, optional if defined at provider level.
+* `rde_type_id` - (Required) The ID of the [RDE Type](/providers/vmware/vcd/latest/docs/data-sources/rde_type) to instantiate.
+* `name` - (Required) The name of the Runtime Defined Entity.
+* `resolve` - (Required) If `true`, the Runtime Defined Entity will be resolved by this provider. If `false`, it won't be
+  resolved and must be either done by an external component or with an update. The Runtime Defined Entity can't be
+  deleted until the input_entity is resolved by either party, unless `resolve_on_removal=true`.
+* `resolve_on_removal` - (Optional) If `true`, the Runtime Defined Entity will be resolved before it gets deleted, to forcefully delete it. Otherwise, destroy will fail if it is not resolved. It is `false` by default.
+* `input_entity` - (Optional) A string that specifies a valid JSON for the RDE. It can be retrieved with functions such as `file`, `templatefile`... Either `input_entity` or `input_entity_url` is required.
+* `input_entity_url` - (Optional) The URL that points to a valid JSON for the RDE. Either `input_entity` or `input_entity_url` is required.
```

Comment:
I think an important thing to mention here is that this will be downloaded on **every read** operation and it will break terraform operations if these contents are no longer present on the remote site. Maybe even hinting that if a user cannot guarantee `input_entity_url` it is safer to use `input_entity`

Reply:
Good point. I added the suggested note and also in the rde_type documentation (which has also this same combination with a different name)